e2etest/clean/clean_distributor_test.go

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package clean
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"sort"
+	"strconv"
+	"testing"
+)
+
+func TestCleanDistributor(t *testing.T) {
+	//set up aws go sdk config
+	awsConfig := aws.Config{
+		Region: aws.String("us-west-2"),
+	}
+	testSession, err := session.NewSession(&awsConfig)
+	if err != nil {
+		t.Fatalf("Error creating session %s", err.Error())
+	}
+	// set up aws go sdk ssm client
+	ssmClient := ssm.New(testSession, &awsConfig)
+
+	//get list of documents with versions
+	documentName := "testAWSDistroOTel-Collector"
+	listDocumentVersionInput := ssm.ListDocumentVersionsInput{Name: &documentName}
+	listDocumentVersionsOutput, err := ssmClient.ListDocumentVersions(&listDocumentVersionInput)
+	if err != nil {
+		t.Fatalf("Error getting document list %s", err.Error())
+	}
+	t.Logf("Document Versions %s", listDocumentVersionsOutput.String())
+	//get document versions
+	var versions []int
+	for i := range listDocumentVersionsOutput.DocumentVersions {
+		//can't delete the default version
+		if *listDocumentVersionsOutput.DocumentVersions[i].IsDefaultVersion {
+			continue
+		}
+		version, err := strconv.Atoi(*listDocumentVersionsOutput.DocumentVersions[i].DocumentVersion)
+		if err != nil {
+			t.Fatalf("Error getting document list %s", err.Error())
+		}
+		versions = append(versions, version)
+	}
+	//sort document versions from oldest to newest
+	sort.Ints(versions[:])
```

Comment:
```suggestion
	sort.Ints(versions)
```

Is there a reason to slice this?  If so, add a comment explaining it.

Reply:
I get a compilation error

Reply:
What error?  It looks like it is possible for `versions` to be a nil slice, though I wouldn't expect a compile error for that.  In any event, it's easy to avoid.  Instead of declaring it like this:

```go
var versions []int
```

do this:

```go
versions := make([]int, 0, len(listDocumentVersionsOutput.DocumentVersions))
```

That makes a new slice of ints with length 0 and capacity to hold as many items as are in the list of document versions.

Reply:
Cannot use 'versions[]' (type int) as the type []int

Reply:
Same issue

Reply:
versions := make([]int, 0, len(listDocumentVersionsOutput.DocumentVersions)) creates a slice if I'm not mistaken since len is not allowed in go array size like it is in java.

Reply:
It would make runtime faster but lets be honest not by much since max size is 25

Reply:
ignore what I said earlier this proper way of doing this.